modules/post/multi/gather/memory_search.rb

```diff
@@ -0,0 +1,99 @@
+# -*- coding: binary -*-
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::MemorySearch
+
+  def initialize(info = {})
+    memory_search_info = initialize_memory_search(
+      info, platforms: %w[linux unix osx windows]
+    )
+
+    super(
+      update_info(
+        memory_search_info,
+        {}
+      )
+    )
+
+    register_options(
+      [
+        ::Msf::OptString.new('PROCESS_NAMES_GLOB', [false, 'Regular expression used to target processes', 'ssh.*']),
+        ::Msf::OptString.new('PROCESS_IDS', [false, 'Comma delimited process ID/IDs to search through']),
+        ::Msf::OptString.new('REGEX', [true, 'Regular expression to search for within memory', 'publickey,password.*']),
+        ::Msf::OptInt.new('MIN_MATCH_LEN', [true, 'The minimum number of bytes to match', 5]),
+        ::Msf::OptInt.new('MAX_MATCH_LEN', [true, 'The maximum number of bytes to match', 127]),
+        ::Msf::OptBool.new('REPLACE_NON_PRINTABLE_BYTES', [false, 'Replace non-printable bytes with "."', true]),
+        ::Msf::OptBool.new('SAVE_LOOT', [false, 'Save the memory matches to loot', true])
+      ]
+    )
+  end
+
+  def get_target_processes
+    raw_target_pids = process_ids || ''
```

Comment:
imo it's a little weird to be calling `process_ids` here when it's defined in the included module but you register the datastore option it uses in this file, same thinking for all the other functions you made for datastore access

Reply:
Changed 👍 There is now only a single module file added so all database options are registered and used in the one file only.